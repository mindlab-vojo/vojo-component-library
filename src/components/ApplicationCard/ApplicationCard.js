import React from 'react'
import PropTypes from 'prop-types'

import Card from '../Card'
import Typography from '../Typography'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import * as ApplicationCard from './ApplicationCard.style'

const ApplicationCardComponent = ({
  currentStep,
  jobTitle,
  imageUrl,
  isClosed,
  onClickCard,
  stepTitle,
  totalSteps,
}) => {

  const renderComponent = (theme) => (
    <Card
      hasPointer
      onClickCard={onClickCard}>
      <ApplicationCard.Wrapper>
        <ApplicationCard.ImageCropper>
          <ApplicationCard.Image 
            isClosed={isClosed}
            src={imageUrl}
          />
        </ApplicationCard.ImageCropper>
        <ApplicationCard.InfoWrapper>
          {
            isClosed ? (
              <>
                <ApplicationCard.InfoText>
                  <Typography
                    fontSize="14px"
                    fontWeight="bolder">
                Candidatura finalizada
                  </Typography>
                </ApplicationCard.InfoText>
                <ApplicationCard.InfoText>
                  <Typography
                    type={'h3'}>
                    {jobTitle}
                  </Typography>
                </ApplicationCard.InfoText>
              </>
            ) : (
              <>
                <ApplicationCard.InfoText>
                  <Typography
                    fontSize="12px">
                    {`Você está na etapa ${currentStep} de ${totalSteps}`}
                  </Typography>
                </ApplicationCard.InfoText>
                <ApplicationCard.InfoText>
                  <Typography
                    type={'h3'}
                    fontWeight={'600'}>
                    {stepTitle}
                  </Typography>
                </ApplicationCard.InfoText>
                <ApplicationCard.InfoText>
                  <Typography
                    type={'subtitle'}
                    tag={'h3'}
                    fontFamily={"'Montserrat', sans-serif"}
                    color={theme.colors.primaryColor}>
                    {jobTitle}
                  </Typography>
                </ApplicationCard.InfoText>
              </>
            )
          }
        </ApplicationCard.InfoWrapper>
      </ApplicationCard.Wrapper>
    </Card>
  )
  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

ApplicationCardComponent.propTypes = {
  currentStep: PropTypes.string,
  jobTitle: PropTypes.string,
  imageUrl: PropTypes.string,
  isClosed: PropTypes.bool,
  onClickCard: PropTypes.func,
  stepTitle: PropTypes.string,
  totalSteps: PropTypes.string,
}

export default ApplicationCardComponent
