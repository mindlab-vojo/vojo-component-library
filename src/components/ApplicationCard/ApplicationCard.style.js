import styled from 'styled-components'

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  width: 100%;
`

export const ImageCropper = styled.div`
  width: 72px;
  height: 72px;
  position: relative;
  overflow: hidden;
  border-radius: 50%;
`

export const Image = styled.img`
  display: inline;
  margin: 0 auto;
  height: 100%;
  width: auto;
  z-index: 1;
  ${props => props.isClosed && `filter: grayscale(100%);`}
`

export const InfoWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  width: 100%;
`