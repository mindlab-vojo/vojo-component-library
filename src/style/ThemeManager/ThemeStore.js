import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore