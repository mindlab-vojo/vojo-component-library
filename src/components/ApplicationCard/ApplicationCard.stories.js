import React from 'react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import ApplicationCard from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const applicationCard = () => {
  const currentStep = text("currentStep", "1")
  const jobTitle = text("jobTitle", "Seletor de produtos para entrega (Shopper)")
  const imageUrl = text("imageUrl", "https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg")  
  const isClosed = boolean("isClosed", false)  
  const onClickCard = action("onClickCard")
  const stepTitle = text("stepTitle", "Procurando empresas")
  const totalSteps = text("totalSteps", "4")
  const fontWeightTitle = text("fontWeightTitle", '700')

  return (
    <ApplicationCard
      currentStep={currentStep}
      imageUrl={imageUrl}
      isClosed={isClosed}
      jobTitle={jobTitle}
      fontWeightTitle={fontWeightTitle}
      onClickCard={onClickCard}
      stepTitle={stepTitle}
      totalSteps={totalSteps}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'ApplicationCard',
  component: applicationCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  