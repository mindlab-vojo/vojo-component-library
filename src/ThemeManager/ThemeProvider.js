import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer