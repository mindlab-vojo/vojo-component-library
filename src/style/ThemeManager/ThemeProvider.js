import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer