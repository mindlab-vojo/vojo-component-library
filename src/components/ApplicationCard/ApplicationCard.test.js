import React from 'react';
import { render } from '@testing-library/react';

import ApplicationCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test ApplicationCard component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <ApplicationCard 
        currentStep={1}
        jobTitle={"Seletor de produtos para entrega (Shopper)"}
        imageUrl={"https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"}
        isClosed={false}
        onClickCard={fakeFn}
        stepTitle={"Procurando empresas"}
        totalSteps={4}
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
