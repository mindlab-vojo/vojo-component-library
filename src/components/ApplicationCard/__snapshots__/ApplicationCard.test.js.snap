// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ApplicationCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm hCpYkD Card"
      >
        <div
          class="sc-AxirZ iTEWhx"
        >
          <div
            class="sc-AxiKw hZuHAH"
          >
            <img
              class="sc-AxhCb fEHPey"
              src="https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"
            />
          </div>
          <div
            class="sc-AxhUy bscpCC"
          >
            <div
              class="sc-AxgMl fFPQpl"
            >
              <p
                class="sc-AxheI oXUPj"
              >
                Você está na etapa 1 de 4
              </p>
            </div>
            <div
              class="sc-AxgMl fFPQpl"
            >
              <p
                class="sc-Axmtr bvymrK"
              >
                Procurando empresas
              </p>
            </div>
            <div
              class="sc-AxgMl fFPQpl"
            >
              <h3
                class="sc-AxmLO bswCGd"
              >
                Seletor de produtos para entrega (Shopper)
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm hCpYkD Card"
    >
      <div
        class="sc-AxirZ iTEWhx"
      >
        <div
          class="sc-AxiKw hZuHAH"
        >
          <img
            class="sc-AxhCb fEHPey"
            src="https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"
          />
        </div>
        <div
          class="sc-AxhUy bscpCC"
        >
          <div
            class="sc-AxgMl fFPQpl"
          >
            <p
              class="sc-AxheI oXUPj"
            >
              Você está na etapa 1 de 4
            </p>
          </div>
          <div
            class="sc-AxgMl fFPQpl"
          >
            <p
              class="sc-Axmtr bvymrK"
            >
              Procurando empresas
            </p>
          </div>
          <div
            class="sc-AxgMl fFPQpl"
          >
            <h3
              class="sc-AxmLO bswCGd"
            >
              Seletor de produtos para entrega (Shopper)
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
